src/pyhf/optimize/autodiff.py

```diff
@@ -1,9 +1,32 @@
+"""Helper Classes for use of automatic differentiation."""
 import scipy
 from .. import get_backend
 
 
 class AutoDiffOptimizerMixin(object):
-    def minimize(self, objective, data, pdf, init_pars, par_bounds, fixed_vals=None):
+    """Mixin Class to build optimizers that use automatic differentiation."""
+
+    def __init__(*args, **kwargs):
+        """Create Mixin for autodiff-based optimizers."""
+        pass
+
+    def minimize(
+        self,
+        objective,
+        data,
+        pdf,
+        init_pars,
+        par_bounds,
+        fixed_vals=None,
+        return_fval=False,
```

Comment:
```suggestion
        return_fit_result=False,
```

Reply:
hm, I feel like the "fit result" is more the bestfit parameters. The value of the objective at the best fit point is more additional info

Reply:
to be fair, you called it "result" in the tests so... not sure... I would prefer `fitted_pars, fitted_vals` instead of `result, fval` or `pars, fval`. `fval` is just too ambiguous for me.